### DISCUSSION TITLE
Encryptions support

### DISCUSSION THREAD

**Discussion by albyrock87** [ericsink/SQLitePCL.raw#596] [Q&A] [answered]
Hi, I need information regarding database encryption.

Looking at [SQLite documentation](https://sqlite.org/com/see.html) it appears the encryption is part of an extension `SEE`.
I wanted to use the `SQLitePCLRaw.bundle_e_sqlite3` package on my .NET MAUI app to have behavior consistency across platforms, but unfortunately I see that package does not bundle this SEE.

I see that `SQLitePCLRaw.bundle_e_sqlcipher` supports encryption but as stated in the `README.md` it is an `unofficial` and `unsupported` package.

So what should I do to have the official and latest `SQLite` builds together with encryption support?

Thanks so much!

**Comment by ericsink** [comment, marked as answer]
We cannot bundle SEE because it is not open source.

As you observed, my builds of the SQLCipher code are unsupported.

If you want an official and supported solution with SQLCipher, you can purchase that from Zetetic.  It is compatible with SQLitePCLRaw.

As an alternative open source solution, you may also want to consider:

https://www.nuget.org/packages/SQLitePCLRaw.bundle_e_sqlite3mc

**Comment by FeliceRu** [comment]
Hi,
i'm using your encryption in my iOS app (with Xamarin) with SQLlite.

Is there any information about the iOS library used for managing encryption?
 
Does the sql-cipher module uses the “Common Crypto” library of the Security Framework of Apple?

If yes, please, could you give an evidence of it?

Thanks.

**Comment by ericsink** [comment]
I don't know if sqlcipher uses Common Crypto on iOS or not.

https://github.com/ericsink/SQLitePCL.raw/wiki/Expectation-setting-for-e_sqlcipher

**Reply by FeliceRu** [reply]
ok.
do you know a way to understand if it uses Common Crypto?
Thanks

**Reply by ericsink** [reply]
You could read the code:

https://github.com/ericsink/cb/tree/master/sqlcipher
